android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java

```diff
@@ -110,50 +111,23 @@ public int onStartCommand(Intent intent, int flags, int startId) {
       IntentFilter filter = new IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED);
       registerReceiver(mReceiver, filter);
 
-      Log.d(TAG, mDisplayUrl);
-      byte[] encodedUrl = AdvertiseDataUtils.encodeUri(mDisplayUrl);
-      if (hasValidUrlLength(encodedUrl.length) && checkAndHandleAsciiUrl(mDisplayUrl)) {
-        // Set the url if we can
-        Log.d(TAG, "valid length");
-        mShareUrl = encodedUrl;
-        broadcastUrl();
-      } else {
-        Log.d(TAG, "needs shortening");
-        UrlShortenerClient.ShortenUrlCallback urlSetter =
-            new UrlShortenerClient.ShortenUrlCallback() {
-          @Override
-          public void onUrlShortened(String newUrl) {
-            Log.d(TAG, "shortening success");
-            mShareUrl = AdvertiseDataUtils.encodeUri(newUrl);
-            broadcastUrl();
-          }
-          @Override
-          public void onError(String oldUrl) {
-            Toast.makeText(getApplicationContext(), getString(R.string.shorten_error),
-                Toast.LENGTH_LONG).show();
-            stopSelf();
-          }
-        };
-        UrlShortenerClient shortenerClient = UrlShortenerClient.getInstance(this);
+      fetchAndHandleUrlFromPWS();
 
-        if (mDisplayUrl.contains("goo.gl")) {
-          // find the site URL and reshorten it since
-          // goo.gl will not reshorten other goo.gl links
-          fetchAndShorten(shortenerClient, urlSetter);
-        } else {
-          shortenerClient.shortenUrl(mDisplayUrl, urlSetter, TAG);
-        }
-      }
       return START_STICKY;
     }
 
-    private void fetchAndShorten(UrlShortenerClient shortenerClient,
-        UrlShortenerClient.ShortenUrlCallback urlSetter) {
-      PwsClient pwsClient = new PwsClient();
-      pwsClient.resolve(Arrays.asList(mDisplayUrl), new PwsResultCallback() {
+    public void fetchAndHandleUrlFromPWS() {
+        PwsClient pwsClient = new PwsClient();
+        pwsClient.resolve(Arrays.asList(mDisplayUrl), new PwsResultCallback() {
         @Override
         public void onPwsResult(PwsResult pwsResult) {
-          shortenerClient.shortenUrl(pwsResult.getSiteUrl(), urlSetter, TAG);
+            mFullUrl = pwsResult.getSiteUrl();
+            if(checkNeedsShortening()) {
+                shortenUrl();
+            }
+            else {
+                broadcastUrl();
+            }
         }
 
         @Override
```

Comment:
github's not letting me write a comment on L135, but is shorten_error really the right text to display here?  Same with PwsResultError.